### PR TITLE
[CDAP-17335] Add Schema editor e2e tests - Error Validation

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds e2e tests for error validations for schema editor.

### CODE REVIEW

File: cdap-ui/cypress/integration/plugin.schemaeditor.spec.ts [hunk 1]
```diff
@@ -760,4 +760,121 @@ describe('Plugin Schema Editor', () => {
...
+  });
+
+  describe('Should perform right client side validation', () => {
+    it('Should correctly validate invalid name', () => {
+      const invalidName = 'invalue@$RWEDSW#@$##!#$@#';
+      cy.visit('/pipelines/ns/default/studio');
+      openProjectionTransform();
+      addField(0, invalidName);
+      cy.get('[data-cy="schema-editor-fieldset-container"]').contains(
+        `invalid field name: invalue@$RWEDSW#@$##!#$@#`
```
Comment (elfenheart) [inline cdap-ui/cypress/integration/plugin.schemaeditor.spec.ts]: nit: string interpolate using `${invalidName}`?

File: cdap-ui/cypress/integration/plugin.schemaeditor.spec.ts [hunk 2]
```diff
@@ -760,4 +760,121 @@ describe('Plugin Schema Editor', () => {
...
+      );
+      removeField(0);
+      addField(0, 'Name', 'record');
+      addField(1, invalidName);
+      cy.get('[data-cy="schema-row-0"] [data-cy="error-icon"]').then((el) => {
+        expect(el).to.have.attr('title', `invalid field name: invalue@$RWEDSW#@$##!#$@#`);
```
Comment (elfenheart) [inline cdap-ui/cypress/integration/plugin.schemaeditor.spec.ts]: here too

**Review by elfenheart — APPROVED** [review]
nit comments, otherwise 👍  LGTM. 

Is the snapshot image intentional? do we want to keep snapshot tests?

**Comment by ajainarayanan** [comment]
> nit comments, otherwise 👍 LGTM.
> 
> Is the snapshot image intentional? do we want to keep snapshot tests?

Yeah the snapshot is intentional. It is to test if we are highlighting the right hierarchy.

**Comment by ajainarayanan** [comment]
Manually cherry-picked the commit to avoid adding Merge commit. Closing this PR.

**Pull request closed without merging**